### PR TITLE
SNOW-654033 add a randomized sleep to handle rmtree error

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-654033

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Fix flaky test for windows environment caused by `rmtree`

### CODE REVIEW

File: tests/integ/scala/test_file_operation_suite.py [hunk 1]
```diff
@@ -24,20 +25,25 @@ def random_alphanumeric_name():
     )
 
 
+def onerror(func, path, exc_info):
+    sleep(1 + random.random() * 4)
```
Comment (sfc-gh-yixie) [inline tests/integ/scala/test_file_operation_suite.py:29]: does it retry automatically after onerror is called?

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_file_operation_suite.py:29]: `onerror` is called after the first error is encountered. So in a way we are giving two tries for `rmtree` to work

Comment (sfc-gh-jdu) [reply tests/integ/scala/test_file_operation_suite.py:29]: so if the second try still fails, it will fail?

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_file_operation_suite.py:29]: yes

**Review by sfc-gh-jdu — APPROVED** [review]

File: tests/integ/scala/test_file_operation_suite.py (hunk 1 above)
Comment (sfc-gh-yixie) [reply tests/integ/scala/test_file_operation_suite.py:29]: I prefer removing the random seconds. If it fails again we have no clue how long time is good.
If we want to wait more time, retry every 1 second and retry 3 or 5 times is better.

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_file_operation_suite.py:29]: how many times should we retry?

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_file_operation_suite.py:29]: I misread your comment. You meant try every 1 sec with 3-5 retries.

**Review by sfc-gh-yixie — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

File: tests/integ/scala/test_file_operation_suite.py [hunk 2]
```diff
@@ -286,6 +286,7 @@ def test_put_stream_with_one_file_twice(session, temp_stage, path1):
     second_result = session.file.put_stream(
         fd, f"{stage_with_prefix}/{file_name}", overwrite=False
     )
+    fd.close()
```
Comment (sfc-gh-aalam) [inline tests/integ/scala/test_file_operation_suite.py:289]: precommit failed again after the earlier "fix" so I check code for all file `open()`s and found one without a `close()`

Comment (sfc-gh-sfan) [reply tests/integ/scala/test_file_operation_suite.py:289]: Should we use `with` in the test, or maybe not using file stream but just `BytesIO`?

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_file_operation_suite.py:289]: We are using `with` is other tests except for two. First, to check if we support two uploads from the same `fd`, and second, to check if we raise error on closed `fd`.

**Pull request merged**